ToyBox/classes/MonkeyPatchin/BagOfPatches/Tweaks.cs

```diff
@@ -273,6 +291,90 @@ static class Tutorial_IsBanned_Patch {
             }
         }
 
+        [HarmonyPatch(typeof(MassLootHelper), nameof(MassLootHelper.GetMassLootFromCurrentArea))]
+        public static class PatchLootEverythingOnLeave_Patch {
+            public static bool Prefix(ref IEnumerable<LootWrapper> __result) {
+                if (!settings.toggleMassLootEverything) return true;
+
+                var all_units = Game.Instance.State.Units.All.Where(w => w.IsInGame);
+                var result_units = all_units.Where(unit => unit.HasLoot).Select(unit => new LootWrapper { Unit = unit }); //unit.IsRevealed && unit.IsDeadAndHasLoot
+
+                var all_entities = Game.Instance.State.Entities.All.Where(w => w.IsInGame);
+                var all_chests = all_entities.Select(s => s.Get<InteractionLootPart>()).Where(i => i?.Loot != Game.Instance.Player.SharedStash).NotNull();
+
+                List<InteractionLootPart> tmp = TempList.Get<InteractionLootPart>();
+
+                foreach (InteractionLootPart i in all_chests) {
+                    //if (i.Owner.IsRevealed
+                    //    && i.Loot.HasLoot
+                    //    && (i.LootViewed
+                    //        || (i.View is DroppedLoot && !i.Owner.Get<DroppedLoot.EntityPartBreathOfMoney>())
+                    //        || i.View.GetComponent<SkinnedMeshRenderer>()))
+                    if (i.Loot.HasLoot) {
+                        tmp.Add(i);
+                    }
+                }
+
+                var result_chests = tmp.Distinct(new MassLootHelper.LootDuplicateCheck()).Select(i => new LootWrapper { InteractionLoot = i });
+
+                __result = result_units.Concat(result_chests);
+#if false   
+                foreach (var loot in __result) // showing inventories from living enemies makes the items invisible (also they can still be looted with the Get All option)
+                {
+                    if (loot.Unit != null)
+                    ;
+                    if (loot.InteractionLoot != null)
+                    ;
+                }
+#endif
+                return false;
+            }
+        }
+
+        [HarmonyPatch(typeof(BlueprintItem), nameof(BlueprintItem.Weight), MethodType.Getter)]
+        public static class BlueprintItem_NoWeight_Patch1 {
```

Comment:
Did you consider just doing BlueprintItem with weight 0.00(some-number-of-0s)1? It's just checking if you're less than or equal to (double) 0.0 and then forcing it to use the type instead if so.